### DISCUSSION TITLE
Using the same request object for multiple test cases

### DISCUSSION THREAD

**Discussion by vinayak-kaladhar** [pactumjs/pactum#106] [Q&A] [answered]
**scenario:** 

I have test case1 , that uses a specific request object(read from file) and makes few key value modifications and uses it for the test cases.

```js
it('Verify creation of HCF with exisiting salesforce id', async () => {
   request.createFacilityWithRequiredFields.salesforceID =  configData.salesforceid
    await pactum.spec()
        .post(endpoints.baseUrl + endpoints.createHCF)
        .withBody(request.createFacilityWithRequiredFields)
        .withHeaders('Accept', 'application/json')
        .withHeaders('Authorization', configData.HCFAuthorization)
        .expectStatus(422)
});
```

I have test case2 , that uses the same specific request object(read from file)

```js
it('Verify creation of surgery facility without surgery related fields ', async () => {
     await pactum.spec()
        .post(endpoints.baseUrl + endpoints.createHCF)
        .withBody(request.createFacilityWithRequiredFields)
        .withHeaders('Accept', 'application/json')
        .withHeaders('Authorization', configData.HCFAuthorization)
        .expectStatus(422)
});
```

But the issue is changes done in test case1 to the request object, gets carried over to test case 2 also.
In this case salesforceID assigned in test case1 gets carried over to test case2

**Questions?**
I believe this is because of async calls, how to mitigate this?
this framework supports only async calls?

**Comment by ASaiAnudeep** [comment, marked as answer]
In general objects are accessed by reference. If you update a value inside the object, it will be reflected at other places. The easiest solution is to have independent objects per test case.

#### Using `clone`

You can create a clone the request object for each test case and modify accordingly. Fortunately `pactum` comes with an inbuilt function to clone objects. Or use `JSON.parse(JSON.stringify(object))`

```js
it('Verify creation of HCF with exisiting salesforce id', async () => {
    const body = pactum.clone(request.createFacilityWithRequiredFields);
    body.salesforceID =  configData.salesforceid
    await pactum.spec()
        .post(endpoints.baseUrl + endpoints.createHCF)
        .withBody(body)
        .withHeaders('Accept', 'application/json')
        .withHeaders('Authorization', configData.HCFAuthorization)
        .expectStatus(422)
});
```

#### Using `DataTemplates`

For a more advanced usage, Data Templates will streamline the way test data can be reused across test cases. Learn more about it [here](https://pactumjs.github.io/#/data-management)

**Reply by vinayak-kaladhar** [reply]
Thanks sai and team, for the swift response.
